paywall/src/__tests__/paywall-builder/build.test.js

```diff
@@ -188,28 +209,67 @@ describe('buildPaywall', () => {
         mockShow.mockImplementation(() => {})
         mockHide = jest.spyOn(iframeManager, 'hide')
         mockHide.mockImplementation(() => {})
+        mockIframe = jest.spyOn(iframeManager, 'getIframe')
+        iframe = {
+          contentWindow: {
+            postMessage,
+          },
+          style: {},
+          remove: jest.fn(),
+        }
+
+        mockIframe.mockImplementation(() => iframe)
         buildPaywall(window, document, fakeLockAddress, blocker)
       })
+
+      it('bails out on error in locked event', () => {
+        expect.assertions(3)
+        mockShow.mockImplementation(() => {
+          throw new Error('thrown')
+        })
+        expect(() => {
+          callbacks.message({ data: POST_MESSAGE_LOCKED })
+        }).toThrow()
+        expect(iframe.remove).toHaveBeenCalled()
+        expect(blocker.remove).toHaveBeenCalled()
+      })
+
       it('triggers show on locked event', () => {
         expect.assertions(2)
         callbacks.message({ data: POST_MESSAGE_LOCKED })
 
-        expect(mockShow).toHaveBeenCalledWith(mockIframeImpl, document)
```

Comment:
this test should never have been passing, but the changes made unearthed its incorrect variable naming